File: app/src/main/java/com/epam/valkaryne/openmovieapp/domain/usecase/SaveQueryHistoryUseCase.java
package com.epam.valkaryne.openmovieapp.domain.usecase;

import com.epam.valkaryne.openmovieapp.core.model.QueryModel;
import com.epam.valkaryne.openmovieapp.domain.QueryRepository;

public class SaveQueryHistoryUseCase {

    private QueryRepository queryRepository;

    public SaveQueryHistoryUseCase(QueryRepository queryRepository) {
        this.queryRepository = queryRepository;
    }

    public void executeUseCase(QueryModel query) {
        queryRepository.saveQuery(query);
    }
}

File: app/src/main/java/com/epam/valkaryne/openmovieapp/domain/usecase/SaveQueryHistoryUseCase.kt
package com.epam.valkaryne.openmovieapp.domain.usecase

import com.epam.valkaryne.openmovieapp.common.QueryModel
import com.epam.valkaryne.openmovieapp.domain.QueryRepository

class SaveQueryHistoryUseCase(private val queryRepository: QueryRepository) {

    fun executeUseCase(query: QueryModel) = queryRepository.saveQuery(query)
}